### DISCUSSION TITLE
How to update plotly to 2.31.1?

### DISCUSSION THREAD

**Discussion by cristiansoto-f** [quarto-dev/quarto-cli#9443] [Q&A]
### Description

Hey 

I'm trying to graph with plotly latest version but I can not properly render data. 

I've tried with 

```
library(htmltools)

price_data_btc <- price_data %>% 
  filter(coin == "BTC-USD")

fig <- plot_ly(price_data_btc, x = ~Date, type = "candlestick",
               open = ~Open, close = ~Close,
               high = ~High, low = ~Low)

plt <- fig  

# create new Plotly JS dependency
newDep <- htmlDependency(name = "plotly-latest", 
                         version = "2.31.1",     
                         src = list(href = "https://cdn.plot.ly"),
                         script = "plotly-2.31.1.min.js")

# add that dependency to plot
#plt$dependencies[[6]] <- newDep

# see what you've got
plt
```

which makes the figure inside the r studio file but when I try to render I get
```
Error in FUN(X[[i]], ...) : 
  Dependency plotly-latest 2.31.1 is not disk-based
Calls: .main ... dependencies_from_render -> html_dependencies_as_string -> lapply -> FUN
Execution halted
```

I've tried with copying the latest plotly min js file into the _freeze folder

```
plt <- fig  

# create new Plotly JS dependency
newDep <- htmlDependency(name = "plotly-main", 
                         version = "2.31.1",     
                         src = list(file = "../../_freeze/site_libs/plotly-main-2.31.1"),
                         script = "plotly-2.31.1.min.js")

# add that dependency to plot
plt$dependencies[[6]] <- newDep
```

and got: 

```
Quitting from lines 218-233 [unnamed-chunk-6] (index.qmd)
Error in `plotlyjsBundleIDX()`:
! Couldn't find the plotlyjs bundle
Backtrace:
  1. global .main()
  2. execute(...)
  3. rmarkdown::render(...)
  4. knitr::knit(knit_input, knit_output, envir = envir, quiet = quiet)
  5. knitr:::process_file(text, output)
     ...
 34. htmlwidgets:::createPayload(x)
 36. plotly:::plotly_build.plotly(instance)
 37. plotly:::verify_partial_bundle(p)
 38. plotly:::plotlyjsBundle(p)
 39. plotly:::plotlyjsBundleIDX(p)
                                                                                                            
Execution halted
```

How should I update the plotly file to be properly rendered?

**Comment by mcanouil** [comment]
That's unrelated to Quarto, I'm afraid.
Please check your tools/packages documentation and repositories.
(Note that here you are using a wrapper R package, so it's it that can eventually upgrade plotjy.js, although this is usually done by maintainers on new releases).
Did you try your code in R?


Note that we ask for a complete and small reproducible example, not snippets of code unrelated to Quarto, in order to help.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.
